File: src/Core.Domain/Entities/Liquidacion.cs
using System.ComponentModel.DataAnnotations.Schema;
using TurnosApp.Core.Domain.Common;
using TurnosApp.Core.Domain.Enums;

namespace TurnosApp.Core.Domain.Entities;

public class Liquidacion : TenantEntity
{
    public int RecursoId { get; set; }

    public DateTime PeriodoDesde { get; set; }
    public DateTime PeriodoHasta { get; set; }
    public DateTime FechaGeneracion { get; set; } = DateTime.UtcNow;

    public EstadoLiquidacion Estado { get; set; } = EstadoLiquidacion.Generada;

    public DateTime? FechaPago { get; set; }
    public int? UsuarioPagoId { get; set; }

    public string? Observaciones { get; set; }

    // ── Calculado, NO persistido — mismo patrón que SesionCaja.MontoEsperadoEfectivo ──
    [NotMapped]
    public decimal MontoBrutoComision => Detalles.Sum(d => d.MontoComisionCalculado);

    [NotMapped]
    public decimal MontoAdelantos => Adelantos.Sum(a => a.Monto);

    [NotMapped]
    public decimal MontoNeto => MontoBrutoComision - MontoAdelantos;

    // Navegación
    public Recurso Recurso { get; set; } = null!;
    public Usuario? UsuarioPago { get; set; }
    public ICollection<LiquidacionDetalle> Detalles { get; set; } = [];
    public ICollection<AdelantoProfesional> Adelantos { get; set; } = [];
}

File: turnos-app-frontend/src/types/Liquidacion.ts
// src/types/Liquidacion.ts

// Coincide con los enums del backend (src/Core.Domain/Enums/TipoComision.cs,
// EstadoLiquidacion.cs, FrecuenciaLiquidacion.cs)
export type TipoComision = 'Porcentaje' | 'MontoFijo';
export type EstadoLiquidacion = 'Generada' | 'Pagada' | 'Anulada';
export type FrecuenciaLiquidacion = 'Semanal' | 'Quincenal' | 'Mensual';

export interface ReglaComision {
  id: number;
  recursoId: number;
  servicioId: number | null;
  servicioNombre: string | null;
  tipo: TipoComision;
  valor: number;
  activo: boolean;
}

export interface ReglaComisionCreateDTO {
  recursoId: number;
  servicioId: number | null;
  tipo: TipoComision;
  valor: number;
}

export interface ReglaComisionUpdateDTO {
  tipo: TipoComision;
  valor: number;
  activo: boolean;
}

export interface AdelantoProfesional {
  id: number;
  recursoId: number;
  monto: number;
  fecha: string;
  concepto: string | null;
  liquidacionId: number | null;
}

export interface AdelantoProfesionalCreateDTO {
  recursoId: number;
  monto: number;
  fecha: string;
  concepto?: string;
}

export interface LiquidacionDetalle {
  id: number;
  turnoId: number;
  turnoFecha: string;
  servicioId: number;
  servicioNombre: string;
  precioBaseAplicado: number;
  tipoComisionSnapshot: TipoComision;
  valorComisionSnapshot: number;
  montoComisionCalculado: number;
}

export interface LiquidacionListItem {
  id: number;
  recursoId: number;
  recursoNombre: string;
  periodoDesde: string;
  periodoHasta: string;
  fechaGeneracion: string;
  estado: EstadoLiquidacion;
  montoBrutoComision: number;
  montoAdelantos: number;
  montoNeto: number;
}

export interface Liquidacion {
  id: number;
  recursoId: number;
  recursoNombre: string;
  periodoDesde: string;
  periodoHasta: string;
  fechaGeneracion: string;
  estado: EstadoLiquidacion;
  fechaPago: string | null;
  usuarioPagoId: number | null;
  usuarioPagoNombre: string | null;
  observaciones: string | null;
  montoBrutoComision: number;
  montoAdelantos: number;
  montoNeto: number;
  detalles: LiquidacionDetalle[];
  adelantos: AdelantoProfesional[];
}

export interface MarcarPagadaLiquidacionDTO {
  observaciones?: string;
}

export interface AnularLiquidacionDTO {
  observaciones?: string;
}
